tmail-backend/event-bus-redis/src/main/java/org/apache/james/events/CleanRedisEventBusService.java

```diff
@@ -0,0 +1,147 @@
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CleanRedisEventBusService {
+    public static class Context {
+        private final AtomicLong totalBindings;
+        private final AtomicLong danglingBindings;
+        private final AtomicLong cleanedBindings;
+
+        public Context() {
+            this.totalBindings = new AtomicLong();
+            this.cleanedBindings = new AtomicLong();
+            this.danglingBindings = new AtomicLong();
+        }
+
+        void increaseScannedBindingsCount() {
+            totalBindings.incrementAndGet();
+        }
+
+        void increaseDanglingBindingsCount(long count) {
+            danglingBindings.addAndGet(count);
+        }
+
+        void increaseCleanedBindingsCount(long count) {
+            cleanedBindings.addAndGet(count);
+        }
+
+        public long getTotalBindings() {
+            return totalBindings.get();
+        }
+
+        public long getDanglingBindings() {
+            return danglingBindings.get();
+        }
+
+        public long getCleanedBindings() {
+            return cleanedBindings.get();
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CleanRedisEventBusService.class);
+
+    private final RedisSetReactiveCommands<String, String> redisSetCommands;
+    private final RedisPubSubReactiveCommands<String, String> redisPubSubCommands;
+    private final RedisKeyReactiveCommands<String, String> redisKeyCommands;
+    private final RoutingKeyConverter routingKeyConverter;
+    private Context context;
+
+    @Inject
+    @Singleton
+    public CleanRedisEventBusService(RedisEventBusClientFactory redisEventBusClientFactory,
+                                     RoutingKeyConverter routingKeyConverter) {
+        this.redisSetCommands = redisEventBusClientFactory.createRedisSetCommand();
+        this.redisPubSubCommands = redisEventBusClientFactory.createRedisPubSubCommand();
+        this.redisKeyCommands = redisEventBusClientFactory.createRedisKeyCommand();
+        this.routingKeyConverter = routingKeyConverter;
+    }
+
+    public Mono<Void> cleanUp() {
+        this.context = new Context();
+
+        return listActiveRedisChannels()
+            .collectList()
+            .flatMap(activeChannels -> listRedisSetKeys()
+                .filter(this::isEventBusBindingKey)
+                .flatMap(key -> listElementsOfSet(key)
+                        .doOnNext(channel -> context.increaseScannedBindingsCount())
+                        .filter(channel -> !activeChannels.contains(channel))
+                        .collectList()
+                        .doOnNext(inactiveChannels -> context.increaseDanglingBindingsCount(inactiveChannels.size()))
+                        .flatMap(deleteBindingsToInactiveChannels(key)),
+                    ReactorUtils.DEFAULT_CONCURRENCY)
+                .then())
+            .doOnSuccess(any -> mdcLogWithContext()
```

Comment:
Should handle case errors because of concurrency. 
eg: I scale up to 3 instances james at a time. 
Then maybe instance B can throws an error when trying delete th key that was done by instance A